### ISSUE TITLE
Clean up test code

### ISSUE THREAD

**Issue by scottbishopdev** [sbishop411/RoverInterviewProject#3]
Most of the code in this repo is pretty darn messy. We should go through and clean it up to my current standards. This issue will be used to track the cleanup of the code used to run our tests.

**Comment by scottbishopdev** [comment]
It should be noted that all of the tests currently implemented are integration tests, not unit tests. This is because they're actually communicating with MongoDB and adding/removing/altering data. We should re-label these tests as such, and create a place where unit tests will be stored as well.

**Comment by scottbishopdev** [comment]
It should also be noted that our tests currently hang on completion. When working with Mocha, this is usually an indication that we haven't properly disposed of all of our connections. I dug around a bit, and I suspect that this is because we're referencing/calling/invoking server.js in the test files, which is opening _it's own_ connection to MongoDB. I discovered this issue when I first attempted to disconnect mongoose in one of the test files, and all subsequent tests timed out, indicating that the server was no longer processing requests for some reason (I think). I've created #5 to fix this.

**Comment by scottbishopdev** [comment]
After a quick look at the test code, I think there there's a number of issues fundamental to it. For starters, all existing tests are integrations tests, not unit tests. Integration tests are great, but the original intent was to build unit tests, so I'd say that I missed the mark on that. Furthermore, I believe that proper integration testing of a Dockerized system like this should also be performed in Docker, as doing so will provide (and test) the same environment for our system components that they'll be running with in prod. Finally, since we're eyeing a conversion to TypeScript for this project, if I'm gonna do a big re-write of our tests, it should probably be in TypeScript. What's more, we _may_ be able to take advantage of the OpenAPI spec we just wrote to automate some of the testing, as OpenAPI _should_ be able to tell some kind of test library what to expect to be returned from specific types of requests submitted to specific endpoints.

In short, I'm throwing this back on the backlog to focus on other things. We'll come back to it later.
